ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -48,10 +48,13 @@ public class DefaultLocalPersistenceService implements LocalPersistenceService {
 
   private final File rootDirectory;
   private final File lockFile;
+  private final File clean;
 
   private FileLock lock;
   private RandomAccessFile rw;
   private boolean started;
+  private final boolean isRootDirectoryExists;
+  private boolean isCleanFileDeleted;
```

Comment:
Wouldn't it make more sense to just have a `clean` variable here that you initialize during start based on what you learn about the state of the disk files. Then your `isClean()` method can be little more than a return of this value (maybe with some guarding to make sure it's been set correctly).